torch_geometric/explain/explanation.py

```diff
@@ -129,3 +131,172 @@ def _apply_masks(
+                if key not in self.node_feat_mask:
+                    status = False
+                    warn_or_raise(
+                        f"Expected a 'node_feat_mask' for node type {key}, "
+                        f"but got none", raise_on_error)
+                elif self.x_dict[key].size() != self.node_feat_mask[key].size(
+                ):
+                    status = False
+                    warn_or_raise(
+                        f"Expected a 'node_feat_mask' of shape "
+                        f"{list(self.x_dict[key].size())} (got shape "
+                        f"{list(self.node_feat_mask[key].size())})",
+                        raise_on_error)
+
+        if 'edge_feat_mask' in self:
+            for key in self.edge_attr_dict.keys():
+                if key not in self.edge_feat_mask:
+                    status = False
+                    warn_or_raise(
+                        f"Expected an 'edge_feat_mask' for edge type {key}, "
+                        f"but got none", raise_on_error)
+                elif self.edge_attr_dict[key].size(
+                ) != self.edge_feat_mask[key].size():
+                    status = False
+                    warn_or_raise(
+                        f"Expected a 'edge_feat_mask' of shape "
+                        f"{list(self.edge_attr_dict[key].size())} (got shape "
+                        f"{list(self.edge_feat_mask[key].size())})",
+                        raise_on_error)
+
+        return status
+
+    def get_explanation_subgraph(self) -> 'HeteroExplanation':
+        r"""Returns the induced subgraph, in which all nodes and edges with
+        zero attribution are masked out."""
+        return self._apply_masks(
+            node_mask={
+                key: value > 0
+                for key, value in self.node_mask.items()
+            } if 'node_mask' in self else None,
+            edge_mask={
+                key: value > 0
+                for key, value in self.edge_mask.items()
+            } if 'edge_mask' in self else None,
+        )
+
+    def get_complement_subgraph(self) -> 'HeteroExplanation':
+        r"""Returns the induced subgraph, in which all nodes and edges with any
+        attribution are masked out."""
+        return self._apply_masks(
+            node_mask={
+                key: value == 0
+                for key, value in self.node_mask.items()
+            } if 'node_mask' in self else None,
+            edge_mask={
+                key: value == 0
+                for key, value in self.edge_mask.items()
+            } if 'edge_mask' in self else None,
+        )
+
+    def _apply_masks(
+        self,
+        node_mask: Optional[Tensor] = None,
+        edge_mask: Optional[Tensor] = None,
+    ) -> 'HeteroExplanation':
+        out = copy.copy(self)
+
+        if edge_mask is not None:
+            for edge_type in self.edge_types:
+                if edge_type not in edge_mask:
+                    continue
```

Comment:
Can we drop this? Because `validate` would fail if a `edge_type` didn't have a mask.